### PR TITLE
Refactor this issuer to use issuer-lib

### PR DESCRIPTION
**Pull request by inteon**
This is the first open-source issuer that will start using this library.
This change greatly simplifies this issuer.

### CODE REVIEW

File: pkg/controllers/signer.go [hunk 1]
```diff
@@ -0,0 +1,205 @@
...
+		RequestId:                     uuid.New().String(),
+		IssuingCertificateAuthorityId: issuerSpec.CertificateAuthorityId,
+	}
+
+	createCertResp, err := casClient.CreateCertificate(ctx, createCertificateRequest)
+	if err != nil {
+		return nil, fmt.Errorf("casClient.CreateCertificate failed: %w", err)
+	}
+
+	// TODO: for backwards compatibility, we should also set the CA field in the certificaterequest
```
Comment (inteon) [inline pkg/controllers/signer.go:126]: This is currently breaking, we have to update issuer-lib to support adding the CA in the certificaterequest.

File: deploy/charts/google-cas-issuer/templates/clusterrole.yaml
Comment (SgtCoDFish) [inline deploy/charts/google-cas-issuer/templates/clusterrole.yaml:1]: suggestion: This file's changes are pretty tricky to review and they seem - at least from an initial glance - to actually be changes in permissions. I'm sure these changes are safe, but maybe it'd be easier to move these to a separate PR with its own justifications?

Or is this coming from issuer-lib and I'm mistaken?

**Review by SgtCoDFish — COMMENTED** [review]
It doesn't look like the e2e test is running, which is really they key in evaluating whether this is safe or not.

IIRC it doesn't run on forks. Could you maybe re-create this PR from a branch on the jetstack/google-cas-issuer repo directly, which I think should trigger the E2E?

**Comment by inteon** [comment]
@SgtCoDFish I just had to add the `ok-to-test` label (see https://github.com/jetstack/google-cas-issuer?tab=readme-ov-file#continuous-integration).
In the future, we probably want to move this repo to the cert-manager org and use prow.

File: deploy/charts/google-cas-issuer/templates/clusterrole.yaml
Comment (inteon) [reply deploy/charts/google-cas-issuer/templates/clusterrole.yaml:1]: I did indeed change both the format and the contents.
Will update the PR and keep the original format.

Comment (inteon) [reply deploy/charts/google-cas-issuer/templates/clusterrole.yaml:1]: Done @SgtCoDFish

**Review by SgtCoDFish — APPROVED** [review]
/lgtm
/approve

Great stuff! I guess next step is to release an alpha version? 🚀
